### ISSUE TITLE
`cb` called multiple times on error

### ISSUE THREAD

**Issue by stcruy** [vsmjs/vsm-dictionary-bioportal#2]
There is a same bug in `getDictInfos()`, `getEntries()`, and `getEntryMatchesForString()` :
In their for-loop, each iteration can make its own call call `cb()`, if an error occurs. See:
```
for (let url of urlArray) {
  ...
  this.request(url, (err, res) => {
    if (err) return cb(err);   //  <---- THAT
```
'THAT' line will get called `urlArray.length` times if each of the request errors for each of the urls.

Solution: set an `answered = true` when an error causes a `cb()` call.
And when `answered` is already `true`, do not call `cb`. Not in the final call, and not when possible next error is received.

**Comment by bblodfon** [comment]
@stcruy Does this relate to the bug we discussed in the email for 'akt1 df' and backspace?

**Comment by bblodfon** [comment]
This also happens in `getEntries` of `vsm-dictionary-uniprot`, and all the functions of `vsm-dictionary-combiner`

**Comment by stcruy** [comment]
@bblodfon 1. No it does not relate to that bug (for which I think I found the cause, and which will be fixed in vsm-box's next release). I only found it while reviewing code to find it.
2. Yes I can assume it does.

**Comment by bblodfon** [comment]
Fixed on commit [`b553b211031b1f941b2bc2fe11896984d876ecc5`](https://github.com/vsmjs/vsm-dictionary-bioportal/commit/b553b211031b1f941b2bc2fe11896984d876ecc5)
